docs/serving/installing-istio.md

```diff
@@ -0,0 +1,139 @@
+  | sed 's/[ \t]*$//' \
+  > ./istio.yaml
+
+kubectl apply -f istio.yaml
+```
+
+### Istio with no Sidecar Injector
+If you don't need sidecar injection in your cluster, you can install an 
+Istio without sidecar injector. Run below command to install the custom Istio.
+```shell
+# A lighter template, with no sidecar injection.
+helm template --namespace=istio-system \
+  --set sidecarInjectorWebhook.enabled=false \
+  --set global.proxy.autoInject=disabled \
+  --set global.omitSidecarInjectorConfigMap=true \
+  --set global.disablePolicyChecks=true \
+  --set prometheus.enabled=false \
+  `# Disable mixer prometheus adapter to remove istio default metrics.` \
+  --set mixer.adapters.prometheus.enabled=false \
+  `# Disable mixer policy check, since in our template we set no policy.` \
+  --set global.disablePolicyChecks=true \
+  `# Set gateway pods to 1 to sidestep eventual consistency / readiness problems.` \
+  --set gateways.istio-ingressgateway.autoscaleMin=1 \
+  --set gateways.istio-ingressgateway.autoscaleMax=1 \
+  `# Set pilot trace sampling to 100%` \
+  --set pilot.traceSampling=100 \
+  install/kubernetes/helm/istio \
+  `# Removing trailing whitespaces to make automation happy` \
+  | sed 's/[ \t]*$//' \
+  > ./istio-lean.yaml
+
+kubectl apply -f istio-lean.yaml
+```
+
+### Install Cluster Local Gateway
+If you want to use the feature of [cluster local route](https://github.com/knative/docs/blob/master/docs/serving/cluster-local-route.md), an extra Istio
+cluster local gateway needs to be added into your cluster. Run below command
+to install the cluster local gateway.
+```shell
+# Generate the extra gateway.
+helm template --namespace=istio-system \
+  --set gateways.custom-gateway.autoscaleMin=1 \
+  --set gateways.custom-gateway.autoscaleMax=1 \
+  --set gateways.custom-gateway.cpu.targetAverageUtilization=60 \
+  --set gateways.custom-gateway.labels.app='cluster-local-gateway' \
+  --set gateways.custom-gateway.labels.istio='cluster-local-gateway' \
+  --set gateways.custom-gateway.type='ClusterIP' \
+  --set gateways.istio-ingressgateway.enabled=false \
+  --set gateways.istio-egressgateway.enabled=false \
+  --set gateways.istio-ilbgateway.enabled=false \
+  install/kubernetes/helm/istio \
+  -f install/kubernetes/helm/istio/example-values/values-istio-gateways.yaml \
+  | sed -e "s/custom-gateway/cluster-local-gateway/g" -e "s/customgateway/clusterlocalgateway/g" \
+  `# Removing trailing whitespaces to make automation happy` \
+  | sed "s/[[:space:]]*$//" \
+  > ./istio-local-gateway.yaml
+
+kubectl apply -f istio-local-gateway.yaml
+```
+
+### Istio with Secret Discovery Service
+[Secret Discovery Service](https://istio.io/docs/tasks/traffic-management/secure-ingress/sds/) is needed if you want to dyanmically update your Gateway 
+with multiple TLS certificates to terminate TLS connection. The below`helm` flag is needed in your `helm` command to enable SDS.
+```
```

Comment:
Can you show the full `helm` command? Showing just this snippet is a bit confusing.

Reply:
Added an example of installing Istio with ingress `SDS` and sidecar injector here.